File: Ywy/JavaScript/level1/ColatzGuess.js
export default class ColatzGuess {
    solution = (num) => {
        let count = 0;
        while (true) {
            if (num === 1) {
                return count;
            }
            count++;
            num = num % 2 === 0 ? Math.floor(num / 2) : num * 3 + 1;

            if (count === 500 || num < 1) {
                return -1;
            }
        }
    }
}

File: Ywy/JAVA/src/level1/ColatzGuess.java
package level1;

public class ColatzGuess {
	public int solution(int n) {
		double x = n * 1.0;
		int count = 0;
		while(true) {
			if (x == 1) {
				return count;
			} else if(count == 500) {
				return -1;
			} 
			count++;
			if (x % 2 == 0) {
				x /= 2;
				continue;
			}
			x *= 3;
			x++;
		}
	}
}
